File: Acupay/Core Forms/PayStub/Entity/PayFrequency.vb
Option Strict On

Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Global.AccuPay.Entity

    <Table("payfrequency")>
    Public Class PayFrequency

        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property RowID As Integer?

        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property Created As Date

        Public Property CreatedBy As Integer?

        <DatabaseGenerated(DatabaseGeneratedOption.Computed)>
        Public Property LastUpd As Date?

        Public Property LastUpdBy As Integer?

        <Column("PayFrequencyType")>
        Public Property Type As String

    End Class

End Namespace

File: Accupay.Data/Entities/PayFrequency.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AccuPay.Data.Entities
{
    [Table("payfrequency")]
    public class PayFrequency
    {
        public const string WeeklyType = "WEEKLY";

        public const string SemiMonthlyType = "SEMI-MONTHLY";

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? RowID { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime Created { get; set; }

        public int? CreatedBy { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime? LastUpd { get; set; }

        public int? LastUpdBy { get; set; }

        [Column("PayFrequencyType")]
        public string Type { get; set; }

        public bool IsWeekly
        {
            get
            {
                return Type.ToUpper() == WeeklyType;
            }
        }

        public bool IsSemiMonthly
        {
            get
            {
                return Type.ToUpper() == SemiMonthlyType;
            }
        }
    }
}